### ISSUE TITLE
Latest ensime wants server jar/pom for reasons?

### ISSUE THREAD

**Issue by parsnips** [ensime/ensime-maven#38]
Running `mvn ensime:generate` fails because it cannot resolve `ensime-server` for the scala version that the maven plugin is running on.  I looked in the pom and couldnt find a dependency listed and `mvn dependency:tree` doesnt show the dep.  

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.584 s
[INFO] Finished at: 2017-02-14T23:02:03+00:00
[INFO] Final Memory: 64M/718M
[INFO] ------------------------------------------------------------------------
[ERROR] Could not find artifact org.ensime:server_2.10:jar:2.0.0-SNAPSHOT in public (http://nexus.banksimple.com/content/repositories/public) -> [Help 1]
```

```
mparsons@snips:~/projects/ensime-maven$ mvn dependency:tree
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=4096M; support was removed in 8.0
Please use CMSClassUnloadingEnabled in place of CMSPermGenSweepingEnabled in the future
[INFO] Scanning for projects...
[WARNING] Failed to retrieve plugin descriptor for org.ensime.maven.plugins:ensime-maven:1.0.0-SNAPSHOT: Plugin org.ensime.maven.plugins:ensime-maven:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact org.ensime.maven.plugins:ensime-maven:jar:1.0.0-SNAPSHOT
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building ensime-maven 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[WARNING] Failed to retrieve plugin descriptor for org.ensime.maven.plugins:ensime-maven:1.0.0-SNAPSHOT: Plugin org.ensime.maven.plugins:ensime-maven:1.0.0-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact org.ensime.maven.plugins:ensime-maven:jar:1.0.0-SNAPSHOT
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ ensime-maven ---
[INFO] org.ensime.maven.plugins:ensime-maven:maven-plugin:1.0.0-SNAPSHOT
[INFO] +- org.apache.maven:maven-project:jar:2.0.11:compile
[INFO] |  +- org.apache.maven:maven-settings:jar:2.0.11:compile
[INFO] |  +- org.apache.maven:maven-profile:jar:2.0.11:compile
[INFO] |  +- org.apache.maven:maven-model:jar:2.0.11:compile
[INFO] |  +- org.apache.maven:maven-artifact-manager:jar:2.0.11:compile
[INFO] |  |  +- org.apache.maven:maven-repository-metadata:jar:2.0.11:compile
[INFO] |  |  \- org.apache.maven.wagon:wagon-provider-api:jar:1.0-beta-2:compile
[INFO] |  +- org.apache.maven:maven-plugin-registry:jar:2.0.11:compile
[INFO] |  +- org.codehaus.plexus:plexus-interpolation:jar:1.1:compile
[INFO] |  +- org.codehaus.plexus:plexus-utils:jar:1.5.6:compile
[INFO] |  +- org.apache.maven:maven-artifact:jar:2.0.11:compile
[INFO] |  \- org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
[INFO] |     \- classworlds:classworlds:jar:1.1-alpha-2:compile
[INFO] +- org.apache.maven:maven-plugin-api:jar:2.0.11:compile
[INFO] +- org.eclipse.aether:aether-api:jar:1.1.0:compile
[INFO] +- org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.1:provided
[INFO] +- org.scala-lang:scala-library:jar:2.10.6:compile
[INFO] +- com.github.scala-incubator.io:scala-io-core_2.10:jar:0.4.3:compile
[INFO] |  \- com.jsuereth:scala-arm_2.10:jar:1.3:compile
[INFO] +- junit:junit-dep:jar:4.10:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.1:test
[INFO] +- org.specs2:specs2_2.10:jar:2.3:test
[INFO] |  +- org.scala-lang:scala-compiler:jar:2.10.3:test
[INFO] |  +- org.scalacheck:scalacheck_2.10:jar:1.10.0:test
[INFO] |  |  +- org.scala-tools.testing:test-interface:jar:0.5:test
[INFO] |  |  \- org.scala-lang:scala-actors:jar:2.10.1:test
[INFO] |  +- junit:junit:jar:4.11:test
[INFO] |  +- org.pegdown:pegdown:jar:1.2.1:test
[INFO] |  |  \- org.parboiled:parboiled-java:jar:1.1.4:test
[INFO] |  |     +- org.parboiled:parboiled-core:jar:1.1.4:test
[INFO] |  |     +- org.ow2.asm:asm:jar:4.1:test
[INFO] |  |     +- org.ow2.asm:asm-tree:jar:4.1:test
[INFO] |  |     +- org.ow2.asm:asm-analysis:jar:4.1:test
[INFO] |  |     \- org.ow2.asm:asm-util:jar:4.1:test
[INFO] |  +- org.specs2:classycle:jar:1.4.1:test
[INFO] |  +- com.chuusai:shapeless_2.10.2:jar:2.0.0-M1:test
[INFO] |  +- org.scalaz:scalaz-core_2.10:jar:7.0.4:test
[INFO] |  +- org.scalaz:scalaz-concurrent_2.10:jar:7.0.4:test
[INFO] |  |  \- org.scalaz:scalaz-effect_2.10:jar:7.0.4:test
[INFO] |  \- org.scala-lang:scala-reflect:jar:2.10.3:test
[INFO] \- org.mockito:mockito-core:jar:1.9.0:test
[INFO]    \- org.objenesis:objenesis:jar:1.0:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.972 s
[INFO] Finished at: 2017-02-14T23:04:39+00:00
[INFO] Final Memory: 13M/309M
[INFO] ------------------------------------------------------------------------
```

cc @amanjpro any idea what I'm missing here?

**Comment by amanjpro** [comment]
Hello @parsnips 

I followed the SBT counterpart, and it includes the server jars. I am not sure if it is really needed by the server, maybe @fommil can answer this.

And I am aware that if a dependency is missing the latest 0.0.6 crashes, it is documents in this issue #35. Which was fixed by PR #36. Is it possible if you clone master, compile and install it:
`mvn install` (it installs is version 1.0.0-SNAPSHOT) and use that to see if it goes through.

I am a little bit surprised that ensime server does not exist for 2.10.6, that is what I was using!
